### DISCUSSION TITLE
6.0.0-beta.3 - Tag checkout on GitHub Actions generates unexpected branch

### DISCUSSION THREAD

**Discussion by prplecake** [GitTools/GitVersion#3653] [Q&A] [answered]
Hi,

I'm running into some familiar issues using GitVersion.MsBuild or GitVersion.Tool 6.0.0-beta.3 and GitHub Actions. Note I am not using the GitVersion action, I'm relying on MsBuild to handle versioning, with GitVersion.Tool used to show output in the GH Actions workflow.

Similar issues: #2838, #3033, #2852, #2074, #2869.

I hadn't been using a `GitVersion.yml` file, but I created one today to specify `ContinousDelivery`, which didn't help.

Other versions I've tried: 5.8.0, 5.10.0, 5.12.0 - each version produced the issue for me, both locally (faking `GITHUB_REF` and `GITHUB_ACTIONS` env vars) and in CI.

Example: https://github.com/prplecake/mastodon-bookmark-sync/actions/runs/5872745163/job/15924827957#step:9:32

**Comment by prplecake** [comment]
Is this an unsupported use case or something? Or what am I doing wrong?

**Reply by asbjornu** [reply]
This is an issue we've had for a long time, unfortunately.

**Reply by HHobeck** [reply, marked as answer]
Is this related to the following issues as well? 
- #3351
- #3614
- #3657

Could you please try this: https://github.com/GitTools/GitVersion/issues/3351#issuecomment-1846661177
